lib/portage/tests/resolver/test_use_dep_defaults.py

```diff
@@ -55,3 +55,195 @@ def testUseDepDefaultse(self):
                 self.assertEqual(test_case.test_success, True, test_case.fail_msg)
         finally:
             playground.cleanup()
+
+    def testUseDepDefaultsDowngrade(self):
+        """
+        Test for bug #917145 where we wrongly prefer a downgrade of sys-apps/systemd
+        instead of either:
+        1) suggesting +kernel-install in a "One of the following packages is required to complete your request" message, or
+        2) autounmask suggesting +kernel-install
+        """
+        installed = {
+            "sys-boot/gnu-efi-3.0.15": {},
+            "sys-apps/systemd-253.11": {
+                "EAPI": 8,
+                "IUSE": "gnuefi",
+                "RDEPEND": "gnuefi? ( sys-boot/gnu-efi )",
+            },
+            "sys-kernel/installkernel-gentoo-7": {
+                "EAPI": 8,
+                "RDEPEND": "!sys-kernel/installkernel-systemd",
+            },
+        }
+        ebuilds = {
+            "sys-boot/gnu-efi-3.0.15": {},
+            "sys-apps/systemd-253.11": {
+                "EAPI": 8,
+                "IUSE": "gnuefi",
+                "RDEPEND": "gnuefi? ( sys-boot/gnu-efi )",
+            },
+            "sys-apps/systemd-254.5": {
+                "EAPI": 8,
+                "IUSE": "kernel-install",
+            },
+            "sys-kernel/installkernel-gentoo-7": {
+                "EAPI": 8,
+                "RDEPEND": "!sys-kernel/installkernel-systemd",
+            },
+            "sys-kernel/installkernel-systemd-2-r4": {
+                "EAPI": 8,
+                "RDEPEND": """
+                    !sys-kernel/installkernel-gentoo
+                    || (
+                        sys-apps/systemd[gnuefi(-)]
+                        sys-apps/systemd[kernel-install(-)]
```

Comment:
Why not swap the order of preference for gnuefi vs. kernel-install flags? Since neither one is satisfied, dep_zapdeps will prefer the one on the left.